CDTDatastore/query/CDTQIndexManager.m

```diff
@@ -214,8 +214,31 @@ - (instancetype)initUsingDatastore:(CDTDatastore *)datastore error:(NSError *__a
  */
 - (NSString *)ensureIndexed:(NSArray * /* NSString */)fieldNames
 {
-    CDTLogError(CDTQ_LOG_CONTEXT, @"-ensureIndexed: not implemented");
-    return nil;
+    // So we don't break semvar w.r.t CDTQIndex and nullablity, we need to
+    // generate the database name here. This does mean there is a risk of a
+    // index name clash if the user is adding indexes from multiple threads.
+
+    NSArray<NSString *> *indexNames = [[self listIndexes] allKeys];
+
+    uint8_t randBytes[20];
+
+    NSString *indexName = nil;
+
+    for (int i = 0; !indexName && i < 200; i++) {
+        arc4random_buf(randBytes, 20);  // not sure I can do this.
```

Comment:
well that comment should have been deleted, I'm not sure if I can/should just pass in 20 the `arc4random_buf` or use `sizeof` (it's been a while since I've worked with C and I'm a little rusty)